### ISSUE TITLE
[Test] Test framework should ignore user's default output and suppress confirm settings. 

### ISSUE THREAD

**Issue by tjprescott** [Azure/azure-cli#2283]
I recorded a test that included a delete operation for a VM. This requires confirmation. I did not put --yes/-y  in that particular command. This caused a hard to diagnose failure on the CI.

The major issue is that the test should have failed to record and play back. It succeeded in both counts.

**Comment by derekbekoe** [comment]
Do you have the config setting 'core', 'disable_confirm_prompt' set on your local machine?

**Comment by derekbekoe** [comment]
Or the env var AZURE_CORE_DISABLE_CONFIRM_PROMPT?

**Comment by tjprescott** [comment]
Hmmmm... yes I do (the config).  We wonder if we can make the test framework ignore that.

**Comment by derekbekoe** [comment]
You can run into similar issues of tests failing in your default output type is not set to json in the config.
Maybe we need a broader issue to make sure when `run_tests` is run locally, it checks some config settings? Or maybe show warnings at the beginning of `run_tests` execution like the following environment variables are set:
```
AZURE_CORE_OUTPUT=...
etc.
```

cc: @troydai

**Comment by tjprescott** [comment]
Or we could simply ignore the config file when running the tests.

**Comment by tjprescott** [comment]
@troydai can you move this into the Testing project?  It won't let me. :-(

**Comment by troydai** [comment]
Yup. The test should capture the environment variable and current setting when the tests are recorded. These should become of part of the tests.

**Comment by JasonRShaver** [comment]
Cleaning u old issues, reopen if this is still active.

**Comment by wurp** [comment]
I just spent the better part of three days debugging an issue caused by this. I just had to update my output entry in c:\User\XXX\.azure\config to be json instead of yaml.

**Comment by wurp** [comment]
> I just spent the better part of three days debugging an issue caused by this. I just had to update my output entry in c:\User\XXX.azure\config to be json instead of yaml.

Symptom:
[gw0] win32 -- Python 3.6.6 E:\git-clones\azure-cli-extensions\env\Scripts\python.exe
env\lib\site-packages\azure_devtools\scenario_tests\preparers.py:41: in _preparer_wrapper
    **kwargs
..\azure-cli\src\azure-cli-testsdk\azure\cli\testsdk\preparers.py:46: in create_resource
    execute(self.cli_ctx, template.format(self.location, name))
..\azure-cli\src\azure-cli-testsdk\azure\cli\testsdk\base.py:200: in __init__
    self._in_process_execute(cli_ctx, command, expect_failure=expect_failure)
..\azure-cli\src\azure-cli-testsdk\azure\cli\testsdk\base.py:263: in _in_process_execute
    raise ex.exception
env\lib\site-packages\knack\cli.py:211: in invoke
    formatter = self.output.get_formatter(output_type)

**Comment by adewaleo** [comment]
@tjprescott, what are your thoughts on this? If we don't have the bandwidth to add this logic we could simply error out or warn the user to use a correct configuration?

**Comment by yonzhan** [comment]
issue triaged.
